src/graphics/gfx_update.cpp

```diff
@@ -1415,22 +1420,28 @@ void UpdateGraphics(bool skipRepaint)
         }
 
 
-        if(LevelEditor)
-        {
-            fBlock = 1;
-            lBlock = numBlock;
-        }
-        else
-        {
-            //fBlock = FirstBlock[int(-vScreenX[Z] / 32) - 1];
-            //lBlock = LastBlock[int((-vScreenX[Z] + vScreen[Z].Width) / 32) + 1];
-            blockTileGet(-vScreenX[Z], vScreen[Z].Width, fBlock, lBlock);
-        }
+//        if(LevelEditor)
+//        {
+//            fBlock = 1;
+//            lBlock = numBlock;
+//        }
+//        else
+//        {
+//            //fBlock = FirstBlock[int(-vScreenX[Z] / 32) - 1];
+//            //lBlock = LastBlock[int((-vScreenX[Z] + vScreen[Z].Width) / 32) + 1];
+//            blockTileGet(-vScreenX[Z], vScreen[Z].Width, fBlock, lBlock);
+//        }
+
+        TreeResult_Sentinel<Block_t> screenBlocks = treeBlockQuery(
+            -vScreenX[Z], -vScreenY[Z],
+            -vScreenX[Z] + vScreen[Z].Width, -vScreenY[Z] + vScreen[Z].Height,
+            SORTMODE_ID);
 
 
 //        For A = fBlock To lBlock 'Non-Sizable Blocks
-        For(A, fBlock, lBlock)
+        for(Block_t* block : screenBlocks)
         {
+            A = block - &Block[1] + 1;
```

Comment:
I think, if there is no index-related formula, then use the block reference directly in the code below. I'll make this by myself later.

Reply:
Looks good. I could also switch the quadtree later to internally use and return `BlockRef_t`s, which will be more cache-friendly and be directly useful for both indexes and references.

Reply:
Good idea :+1: